Source/WebCore/html/ImageDocument.cpp

```diff
@@ -147,6 +147,9 @@ void ImageDocument::updateDuringParsing()
     if (!m_imageElement)
         createDocumentStructure();
 
+    if (!document().frame())
```

Comment:
I believe `document()` returns `this` here given this `this` is an ImageDocument (which is a Document). Therefore, we can probably use `if (!frame())`

Reply:
yes, i feel the same and was confused by the usage of `this` in other functions from this ImageDocument. addressed in latest patch.